File: src/wpsr/parsers/rdf/predicates/Predicate.java
package wpsr.parsers.rdf.predicates;

import wpsr.parsers.dom.DOMCandidateManager;
import wpsr.parsers.dom.DOMProcessor;
import wpsr.parsers.rdf.query.IRDFResource;
import wpsr.parsers.rdf.query.IRDFStatement;

public abstract class Predicate {

    private DOMProcessor proc;

    public Predicate(DOMProcessor proc) {
	this.proc = proc;
    }

    public DOMProcessor getDOMProcessor() {
	return proc;
    }

    public void analyze(IRDFStatement stmt, DOMCandidateManager manager) {
        analyze(stmt, null, manager);
    }

    public abstract void analyze(IRDFStatement stmt, IRDFResource root, DOMCandidateManager manager);
}

File: src/wpsr/parsers/rdf/predicates/Predicate.ts
///<reference path="./../../dom/DOMCandidateManager.ts" />
///<reference path="./../../dom/DOMProcessor.ts" />
///<reference path="./../query/IRDFResource.d.ts" />
///<reference path="./../query/IRDFStatement.d.ts" />

// TODO: Predicate should be an abstract class or an interface
module wpsr.parsers.rdf.predicates {
  abstract class Predicate {
    private processor: DOMProcessor;
    constructor(processor: DOMProcessor) {
      this.processor = processor;
    }
    public getDOMProcessor(): DOMProcessor {
      return this.processor;
    }
    public analyze(stmt: IRDFStatement, manager: DOMCandidateManager): void {
      analyze(stmt, null, manager);
    }
    public analyze(stmt: IRDFStatement, root: IRDFResource, manager: DOMCandidateManager): void {
    };
  }
}
